pkg/common/resources/tracked_resources.go

```diff
@@ -103,3 +103,46 @@ func (tr *TrackedResource) AggregateTrackedResource(instType string,
 	}
 	tr.TrackedResourceMap[instType] = aggregatedResourceTime
 }
+
+func EqualsTracked(left, right *TrackedResource) bool {
+	if left == right {
+		return true
+	}
+
+	if left == nil || right == nil {
+		return false
+	}
+
+	equals := true
+	for k, v := range left.TrackedResourceMap {
+		inner, ok := right.TrackedResourceMap[k]
+		if !ok {
+			return false
+		}
+
+		equals = equals && equalsMapContents(v, inner)
+	}
+
+	return equals
```

Comment:
Slightly better without `equals` variable:

```
	for k, v := range left.TrackedResourceMap {
		inner, ok := right.TrackedResourceMap[k]
		if !ok {
			return false
		}

		if !equalsMapContents(v, inner) {
			return false
		}
	}

	return true
```

Reply:
Nice improvement! Fixed in commit 9b3c237